### PR TITLE
Add compatibility for Lit 2.0

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

This updates our code base to the new API of Lit 2.0 using the forward compatibility of lit-element 2.5 and lit-html 1.4

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [X] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/ha-panel-lovelace.ts [hunk 1]
```diff
@@ -103,9 +97,9 @@ class LovelacePanel extends LitElement {
   }
 
   protected render(): TemplateResult | void {
-    const state = this._state!;
+    const entityState = this._state!;
```
Comment (ludeeus) [inline src/panels/lovelace/ha-panel-lovelace.ts:100]: This is not a entity

**Review by ludeeus — APPROVED** [review]

File: src/panels/lovelace/ha-panel-lovelace.ts (hunk 1 above)
Comment (bramkragten) [reply src/panels/lovelace/ha-panel-lovelace.ts:100]: Define entity 😜

👍🏻 I'll rename

**Review by balloob — APPROVED** [review]

File: src/components/ha-markdown-element.ts [hunk 2]
```diff
@@ -1,15 +1,22 @@
...
   @property() public content?;
 
   @property({ type: Boolean }) public allowSvg = false;
 
   @property({ type: Boolean }) public breaks = false;
 
+  public connectedCallback() {
+    super.connectedCallback();
+    if (!this.shadowRoot) {
+      this.attachShadow({ mode: "open" });
```
Comment (balloob) [inline src/components/ha-markdown-element.ts:16]: This didn't have a shadow root on purpose. `<ha-markdown>` is the one with shadow dom. This element was just to render the markdown. 

This causes `<ha-markdown>` styling to not be applied

image

Comment (bramkragten) [reply src/components/ha-markdown-element.ts:16]: Ok, will fix

Comment (balloob) [reply src/components/ha-markdown-element.ts:16]: https://github.com/home-assistant/frontend/pull/9187

**Pull request merged**